File: BaekJoon Online Judge/BFS(Breadth-First-Search)/#15240 Paint bucket/PaintBucket.java
import java.io.*;
import java.util.*;

class Pair{
    int x_, y_;

    public Pair(int x_, int y_) {
        this.x_ = x_;
        this.y_ = y_;
    }
}

public class Main {

    /**
     * @param args the command line arguments
     */
    public static int R,C;
    public static StringBuilder[] image;
    public static int[] dx = {-1, 0, 1, 0};
    public static int[] dy = { 0,-1, 0, 1};

    
    public static void bfs(Pair point, char origin_color, char color){
        Queue<Pair> q = new LinkedList<>();
        
        q.add(point);
        
        while(!q.isEmpty()){
            Pair cur = q.poll();
                    
            for (int i = 0; i < 4; i++) {
                Pair next = new Pair(cur.x_ + dx[i], cur.y_ + dy[i]);
                if(next.x_ < 0 || next.y_ < 0 || next.x_ >= R || next.y_ >= C) continue;
                char next_char = image[next.x_].charAt(next.y_);
            
                if(next_char == color) continue;
                if(next_char != origin_color) continue;
            
                image[next.x_].setCharAt(next.y_, color);
                q.offer(next);
            }
        }

    }
    
    public static void main(String[] args) throws IOException {
        // TODO code application logic here
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String[] input = br.readLine().split(" ");
        
        R = Integer.parseInt(input[0]);
        C = Integer.parseInt(input[1]);
        
        image = new StringBuilder[R];
        
        for (int i = 0; i < R; i++) image[i] = new StringBuilder(br.readLine());
        
        input = br.readLine().split(" ");
        
        int x = Integer.parseInt(input[0]);
        int y = Integer.parseInt(input[1]);
        char color = (char)(Integer.parseInt(input[2]) + '0');
        char origin_color = image[x].charAt(y);
        
        image[x].setCharAt(y, color);
        bfs(new Pair(x, y), origin_color, color);
        
        for (StringBuilder str : image) bw.write(str.toString() + "\n");
        bw.flush();
    }
}

File: BaekJoon Online Judge/BFS(Breadth-First-Search)/#15240 Paint bucket/PaintBucket.c
#include <stdio.h>

typedef struct Pair{
    int x_;
    int y_;
} Pair;

int R,C;
int dx[] = {-1, 0, 1, 0};
int dy[] = { 0,-1, 0, 1};
char image[1000][1000];

void dfs(Pair point, char origin_color, char color){
    image[point.x_][point.y_] = color;

    for (int i = 0; i < 4; i++) {
        Pair next;
        next.x_ = point.x_ + dx[i];
        next.y_ = point.y_ + dy[i];
        
        if(next.x_ < 0 || next.y_ < 0 || next.x_ >= R || next.y_ >= C || image[next.x_][next.y_] != origin_color) continue;
            
        dfs(next, origin_color, color);
    }
}

int main()
{
    scanf("%d %d", &R, &C);
    
    char buffer;
    scanf("%c", &buffer);
    
    for(int i = 0; i < R; i++) scanf("%s", image[i]);
    
    Pair point;
    char color;
    scanf("%d %d %c", &point.x_, &point.y_, &color);
    
    dfs(point, image[point.x_][point.y_], color);
    
    for(int i = 0; i < R; i++) printf("%s\n", image[i]);
    
    return 0;
}
